pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java

```diff
@@ -70,6 +70,8 @@ private RelDataType toRelDataType(FieldSpec fieldSpec) {
         return createSqlType(SqlTypeName.VARCHAR);
       case BYTES:
         return createSqlType(SqlTypeName.VARBINARY);
+      case BIG_DECIMAL:
```

Comment:
I feel adding a new data type needs a lot more thorough testing. Say we make sure func calls work on this, filter/transform work on this etc.

It is better to say we don't support this type instead of we support this sometimes.

Reply:
i originally though so too but this one is specific to null default handling. so I am ok with this.

Reply:
Hmm.. Why is this specific to null default handling?

Reply:
I was planning on adding a lot more tests when I get to the mathematical operators. It's also not a new type, V1 supports it `BIG_DECIMAL` this was just necessary to wire v2 with v1.

Reply:
also I think Rong was referring to `FieldSpec.java` where I added:
```
            case BIG_DECIMAL:
               return DEFAULT_DIMENSION_NULL_VALUE_OF_BIG_DECIMAL;
```
this change that you're commenting on was the TypeFactory that maps Calcite types to Pinot types.

Reply:
SG

Reply:
yeah, I assumed @61yao was talking also about the general supports, and not just the TypeFactory which is used exclusively for V2.